### PR TITLE
docs: corrected grammatical errors in the docs

### PR DESCRIPTION
**Pull request by jyotisachdeva57**
- Corrected grammatical mistakes.
- Added missing type annotations.

### CODE REVIEW

File: doc/licenses/apache_jna.txt [hunk 1]
```diff
@@ -88,7 +88,7 @@ Apache 2 license:
       or a Contribution incorporated within the Work constitutes direct
       or contributory patent infringement, then any patent licenses
       granted to You under this License for that Work shall terminate
-      as of the date such litigation is filed.
+      as of the date, such litigation is filed.
```
Comment (SethTisue) [inline doc/licenses/apache_jna.txt]: this is somebody else's license file, so it's important it to leave it verbatim and not make any edits in our copy

File: spec/01-lexical-syntax.md [hunk 2]
```diff
@@ -9,17 +9,17 @@ chapter: 1
 Scala programs are written using the Unicode Basic Multilingual Plane
 (_BMP_) character set; Unicode supplementary characters are not
 presently supported.  This chapter defines the two modes of Scala's
-lexical syntax, the Scala mode and the _XML mode_. If not
+lexical syntax, the Scala mode, and the _XML mode_. If not
 otherwise mentioned, the following descriptions of Scala tokens refer
-to _Scala mode_, and literal characters ‘c’ refer to the ASCII fragment
+to _Scala mode_, and literal characters ‘c’ refers to the ASCII fragment
```
Comment (SethTisue) [inline spec/01-lexical-syntax.md]: I think "refer" is correct here. "characters" is plural. Perhaps a rewording would improve clarity.

File: spec/01-lexical-syntax.md [hunk 3]
```diff
@@ -47,7 +47,7 @@ hexDigit      ::= ‘0’ | … | ‘9’ | ‘A’ | … | ‘F’ | ‘a’ |
 ```
 
 There are three ways to form an identifier. First, an identifier can
-start with a letter which can be followed by an arbitrary sequence of
+start with a letter that can be followed by an arbitrary sequence of
```
Comment (SethTisue) [inline spec/01-lexical-syntax.md]: perhaps better: "start with a letter, followed by"

File: spec/01-lexical-syntax.md [hunk 4]
```diff
@@ -71,7 +71,7 @@ The rules for pattern matching further distinguish between
 _variable identifiers_, which start with a lower case letter
 or `_`, and _constant identifiers_, which do not.
 
-For this purpose, lower case letters doesn't only include a-z,
+For this purpose, lower case letters don't only include a-z,
```
Comment (SethTisue) [inline spec/01-lexical-syntax.md]: "include not only" would flow a bit better, I think

File: spec/02-identifiers-names-and-scopes.md [hunk 5]
```diff
@@ -65,7 +65,7 @@ of the referenced entity.
 A reference to a qualified (type- or term-) identifier $e.x$ refers to
 the member of the type $T$ of $e$ which has the name $x$ in the same
 namespace as the identifier. It is an error if $T$ is not a [value type](03-types.html#value-types).
-The type of $e.x$ is the member type of the referenced entity in $T$.
+The type of $e.x$ is the member type of the referenced entity in`` $T$.
```
Comment (SethTisue) [inline spec/02-identifiers-names-and-scopes.md]: inadvertent change?

**Review by SethTisue — CHANGES_REQUESTED** [review]
thank you for this! I left some specific feedback comments

File: spec/02-identifiers-names-and-scopes.md (hunk 5 above)
Comment (jyotisachdeva57) [reply spec/02-identifiers-names-and-scopes.md]: Yes, reverted it.

File: spec/01-lexical-syntax.md (hunk 4 above)
Comment (jyotisachdeva57) [reply spec/01-lexical-syntax.md]: Changed as per suggestion.

File: spec/01-lexical-syntax.md (hunk 3 above)
Comment (jyotisachdeva57) [reply spec/01-lexical-syntax.md]: Agreed. Changed it.

File: spec/01-lexical-syntax.md (hunk 2 above)
Comment (jyotisachdeva57) [reply spec/01-lexical-syntax.md]: Yes. You are right. Changed it back.

File: doc/licenses/apache_jna.txt (hunk 1 above)
Comment (jyotisachdeva57) [reply doc/licenses/apache_jna.txt]: Sure. Reverted the changes from this file.

File: spec/01-lexical-syntax.md [hunk 6]
```diff
@@ -71,7 +71,7 @@ The rules for pattern matching further distinguish between
 _variable identifiers_, which start with a lower case letter
 or `_`, and _constant identifiers_, which do not.
 
-For this purpose, lower case letters doesn't only include a-z,
+For this purpose, lower case letters include not only include a-z,
```
Comment (diesalbla) [inline spec/01-lexical-syntax.md]: There are two `include` words in here...

**Review by diesalbla — APPROVED** [review]

File: spec/01-lexical-syntax.md (hunk 6 above)
Comment (jyotisachdeva57) [reply spec/01-lexical-syntax.md]: Removed extra include.

**Review by dwijnand — APPROVED** [review]

**Review by dwijnand — APPROVED** [review]
Thanks!

**Pull request merged**